patches/net/minecraft/world/item/FireChargeItem.java.patch

```diff
@@ -0,0 +1,11 @@
+--- a/net/minecraft/world/item/FireChargeItem.java
++++ b/net/minecraft/world/item/FireChargeItem.java
+@@ -33,7 +_,7 @@
+         BlockPos blockpos = p_41204_.getClickedPos();
+         BlockState blockstate = level.getBlockState(blockpos);
+         boolean flag = false;
+-        if (!CampfireBlock.canLight(blockstate) && !CandleBlock.canLight(blockstate) && !CandleCakeBlock.canLight(blockstate)) {
++        if (blockstate.getBlock() instanceof net.neoforged.neoforge.common.SpecialLightableBlock && !((net.neoforged.neoforge.common.SpecialLightableBlock) blockstate.getBlock()).canLight(blockstate)) {
```

Comment:
wrong logic, instead of catching unlightable blocks and lightable blocks that refuse, it only catches the latter. Should be `!(instanceof) || !canLight()`

I don't see the else, but I suspect it also sets LIT